contracts/src/CrossChainApplications/examples/NativeTokenBridge/NativeTokenDestination.sol

```diff
@@ -26,41 +29,101 @@ import {IAllowList} from "@avalabs/subnet-evm-contracts@1.2.0/contracts/interfac
+     */
     uint256 public constant TRANSFER_NATIVE_TOKENS_REQUIRED_GAS = 100_000;
+
+    /**
+     * @notice Estimated gas needed for a burn call to execute successfully on the source chain.
+     */
     uint256 public constant REPORT_BURNED_TOKENS_REQUIRED_GAS = 100_000;
+
+    /**
+     * @notice ID of the paired blockchain containing the token source contract.
+     */
     bytes32 public immutable sourceBlockchainID;
+
+    /**
+     * @notice Address of the paired token source contract.
+     */
     address public immutable nativeTokenSourceAddress;
-    // The first `initialReserveImbalance` tokens sent to this subnet will not be minted.
-    // `initialReserveImbalance` should be constructed to match the initial token supply of this subnet.
-    // This means tokens will not be minted until the source contact is collateralized.
+
+    /**
+     * @notice Initial reserve imbalance that must be collateralized on the source before minting.
+     *
+     * @dev The first `initialReserveImbalance` tokens sent to this subnet will not be minted.
+     * `initialReserveImbalance` should be constructed to match the initial token supply of this subnet.
+     * This means tokens will not be minted until the source contact is collateralized.
+     */
     uint256 public immutable initialReserveImbalance;
+
+    /**
+     * @notice Tokens will not be minted until this value is 0 meaning the source contact is collateralized.
+     */
     uint256 public currentReserveImbalance;
+
+    /**
+     * @notice Total number of tokens minted by this contract through the native minter precompile.
+     */
     uint256 public totalMinted;
 
+    /**
+     * @notice tokenMultiplier allows this contract to scale the number of tokens it sends/receives to/from
+     * the source chain.
+     *
+     * @dev This can be used to normalize the number of decimals places between the tokens on
+     * the two subnets. Is calculated as 10^d, where d is decimalsShift specified in the constructor.
+     */
+    uint256 public immutable tokenMultiplier;
+
+    /**
+     * @notice If multiplyOnReceive is true, the raw token amount value will be multiplied by `tokenMultiplier` when tokens
+     * are transferred from the source chain into this destination chain, and divided by `tokenMultiplier` when
+     * tokens are transferred from this destination chain back to the source chain. This is intended
+     * when the "decimals" value on the source chain is less than the native EVM denomination of 18.
+     * If multiplyOnReceive is false, the raw token amount value will be divided by `tokenMultiplier` when tokens
+     * are transferred from the source chain into this destination chain, and multiplied by `tokenMultiplier` when
+     * tokens are transferred from this destination chain back to the source chain.
+     */
+    bool public immutable multiplyOnReceive;
+
     constructor(
```

Comment:
might want some doc about `decimalShift`

Reply:
I see the `README` has a more detailed explanation, can you add a basic Natspec `@dev` for the overall constructor function.